tests/io/hdf5_test.py

```diff
@@ -30,6 +31,12 @@ def check_roundtrip(obj):
 
 eigen_2d = sc.matrices(dims=['x'], values=np.random.rand(4, 3, 3))
```

Comment:
Unrelated, but can you improve these names (to match the ones you used for the new dtypes)? The fact that `Eigen` is used internally is irrelevant?